tests/integration/ha/test_backups.py

```diff
@@ -0,0 +1,570 @@
+import uuid
+from typing import Dict
+
+import boto3
+import pytest
+from charms.opensearch.v0.opensearch_backups import S3_REPOSITORY
+from pytest_operator.plugin import OpsTest
+
+from ..ha.continuous_writes import ContinuousWrites
+from ..ha.test_horizontal_scaling import IDLE_PERIOD
+from ..helpers import (
+    APP_NAME,
+    MODEL_CONFIG,
+    SERIES,
+    get_leader_unit_id,
+    get_leader_unit_ip,
+    http_request,
+    run_action,
+)
+from ..helpers_deployments import get_application_units, wait_until
+from ..tls.test_tls import TLS_CERTIFICATES_APP_NAME
+from .helpers import (
+    app_name,
+    assert_continuous_writes_consistency,
+    assert_continuous_writes_increasing,
+    assert_cwrites_backup_consistency,
+    create_backup,
+    list_backups,
+    restore,
+    start_and_check_continuous_writes,
+)
+from .helpers_data import index_docs_count
+
+logger = logging.getLogger(__name__)
+
+S3_INTEGRATOR = "s3-integrator"
+S3_INTEGRATOR_CHANNEL = "latest/edge"
+TIMEOUT = 10 * 60
+BackupsPath = f"opensearch/{uuid.uuid4()}"
+
+
+# We use this global variable to track the current relation of:
+#    backup-id <-> continuous-writes index document count
+# We use this global variable then to restore each backup on full DR scenario.
+cwrites_backup_doc_count = {}
+
+
+@pytest.fixture(scope="session")
+def cloud_configs(
+    github_secrets: Dict[str, str], microceph: Dict[str, str]
+) -> Dict[str, Dict[str, str]]:
+    # Figure out the address of the LXD host itself, where tests are executed
+    # this is where microceph will be installed.
+    ip = subprocess.check_output(["hostname", "-I"]).decode().split()[0]
+    results = {
+        "microceph": {
+            "endpoint": f"http://{ip}",
+            "bucket": microceph.bucket,
+            "path": BackupsPath,
+            "region": "default",
+        },
+    }
+    if "AWS_ACCESS_KEY" in github_secrets:
+        results["aws"] = {
+            "endpoint": "https://s3.amazonaws.com",
+            "bucket": "data-charms-testing",
+            "path": BackupsPath,
+            "region": "us-east-1",
+        }
+    return results
+
+
+@pytest.fixture(scope="session")
+def cloud_credentials(
+    github_secrets: Dict[str, str], microceph: Dict[str, str]
+) -> Dict[str, Dict[str, str]]:
+    """Read cloud credentials."""
+    results = {
+        "microceph": {
+            "access-key": microceph.access_key_id,
+            "secret-key": microceph.secret_access_key,
+        },
+    }
+    if "AWS_ACCESS_KEY" in github_secrets:
+        results["aws"] = {
+            "access-key": github_secrets["AWS_ACCESS_KEY"],
+            "secret-key": github_secrets["AWS_SECRET_KEY"],
+        }
+    return results
+
+
+@pytest.fixture(scope="session", autouse=True)
+def remove_backups(
+    # ops_test: OpsTest,
+    cloud_configs: Dict[str, Dict[str, str]],
+    cloud_credentials: Dict[str, Dict[str, str]],
+):
+    """Remove previously created backups from the cloud-corresponding bucket."""
```

Comment:
I believe you can set up your cloud to autoclear every day

Reply:
Maybe, but these are shared across our teams.